api/data_explorer/controllers/facets_controller.py

```diff
@@ -28,25 +32,59 @@ def _get_bucket_interval(facet):
         return _get_bucket_interval(facet.nested_facet)
 
 
-def facets_get(filter=None):  # noqa: E501
+def _process_extra_facets(extra_facets):
+    es = Elasticsearch(current_app.config['ELASTICSEARCH_URL'])
+
+    es_facets = OrderedDict()
+    ui_facets = OrderedDict()
+
+    # Update the extra facets dicts in the config, only if the extra facets change.
+    if extra_facets:
+        for elasticsearch_field_name in extra_facets:
+            if elasticsearch_field_name:
```

Comment:
Hmm, I had to add this because in a particular sequence of steps from the UI, when there are no extra facets, an array with an empty string gets passed. I'll show you this when I come in tomorrow.